src/test/java/uk/nhs/hee/tis/trainee/credentials/api/PlacementCredentialTest.java

```diff
@@ -0,0 +1,90 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.trainee.credentials.api;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import uk.nhs.hee.tis.trainee.credentials.TestJwtUtil;
+import uk.nhs.hee.tis.trainee.credentials.dto.CredentialType;
+import uk.nhs.hee.tis.trainee.credentials.mapper.CredentialMetadataMapper;
+import uk.nhs.hee.tis.trainee.credentials.mapper.CredentialMetadataMapperImpl;
+import uk.nhs.hee.tis.trainee.credentials.service.CredentialMetadataService;
+
+@ContextConfiguration(classes = {CredentialMetadataMapper.class,
+    CredentialMetadataMapperImpl.class})
+@ExtendWith(SpringExtension.class)
+@WebMvcTest(PlacementResource.class)
+class PlacementCredentialTest {
+
+  @Autowired
+  private MockMvc mockMvc;
+  @Autowired
+  private MappingJackson2HttpMessageConverter jacksonMessageConverter;
+  @Mock
+  private CredentialMetadataService service;
+
+
+  /**
+   * Set up mocks before each test.
+   */
+  @BeforeEach
+  void setup() {
+    PlacementResource placementResource = new PlacementResource(service);
+    this.mockMvc = MockMvcBuilders.standaloneSetup(placementResource)
+        .setMessageConverters(jacksonMessageConverter)
+        .build();
+  }
+
+  @Test
+  void testSuccessfullyCallTheService() throws Exception {
```

Comment:
You should mock the service call result and verify that the controller implementation is processing the result correctly. Separate tests for empty, single and multiple results will ensure that we're handling each scenario. We also know that a `NoSuchElementException` is possible, so we should verify behaviour for that too.

Currently I could rewrite the API return like below and it would pass your test without doing anything that we want.

```java
return ResponseEntity.ok().build();
```